### PR TITLE
Sb 28325

### PR DESCRIPTION
**Pull request by vishwanath1004**

### CODE REVIEW

File: src/app.scss [hunk 1]
```diff
@@ -2307,4 +2319,11 @@ app-metadata-details{
         text-transform: none;
     }
 }
-}
\ No newline at end of file
+.attachment-delete-alert{
+  background-color: red;
+  ion-button{
+    color:#adc3dd;
+    --color:green;
```
Comment (kiranharidas187) [inline src/app.scss:2326]: check the color

File: src/app/manage-learn/core/services/attachment/attachment.service.ts [hunk 2]
```diff
@@ -116,6 +118,7 @@ export class AttachmentService {
         };
 
         this.presentToast(this.texts["FRMELEMNTS_MSG_SUCCESSFULLY_ATTACHED"], "success");
+        this.actionEvent.emit(data);
```
Comment (kiranharidas187) [inline src/app/manage-learn/core/services/attachment/attachment.service.ts:121]: check with flow discussed

File: src/app/manage-learn/core/services/attachment/attachment.service.ts [hunk 3]
```diff
@@ -203,4 +207,26 @@ export class AttachmentService {
...
+      break;
+      case 'openGallery':
+       this.takePicture(this.camera.PictureSourceType.PHOTOLIBRARY);
+      break;
+      case 'openFiles':
+        this.openFile();
+      break;
+    }
+  }
+  async openLinkModal(){
```
Comment (kiranharidas187) [inline src/app/manage-learn/core/services/attachment/attachment.service.ts:225]: rm

File: src/app/manage-learn/project/add-file/add-file.page.ts [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+  ) {
+
+    routerParams.params.subscribe(urlParams => {
+      this.projectId = urlParams.id;
+      this.getProject();
+    })
+    routerParams.queryParams.subscribe(params => {
+      this.description = params.taskId ? actions.TASK_FILE_DESCRIPTION.label : actions.PROJECT_FILE_DESCRIPTION.label;
+      this.taskId = params.taskId;
+      this.viewOnlyMode = params.viewOnlyMode;
```
Comment (kiranharidas187) [inline src/app/manage-learn/project/add-file/add-file.page.ts:57]: rm

File: src/app/manage-learn/project/add-file/add-file.page.ts [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    });
+  
+  }
+  getProject() {
+    this.db.query({ _id: this.projectId }).then(
+      (success) => {
+        if (success?.docs.length) {
+          this.project = success.docs[0];
+        }
+        this.projectCopy = JSON.parse(JSON.stringify(this.project));
```
Comment (kiranharidas187) [inline src/app/manage-learn/project/add-file/add-file.page.ts:84]: { ... this.project}

File: src/app/manage-learn/project/project-details/project-details.component.ts [hunk 6]
```diff
@@ -262,6 +262,7 @@ export class ProjectDetailsComponent implements OnInit {
     this.db.update(this.projectDetails).then(success => {
       this.projectDetails._rev = success.rev;
       this.taskCount = this.utils.getTaskCount(this.projectDetails);
+      this.projectDetails.status = statusType.submitted ?  this.router.navigate([`${RouterLinks.PROJECT}/${RouterLinks.ADD_FILE}`,this.projectDetails._id]) :'';
```
Comment (kiranharidas187) [inline src/app/manage-learn/project/project-details/project-details.component.ts:265]: check

File: src/app/manage-learn/project/task-view/task-view.page.ts [hunk 7]
```diff
@@ -317,9 +318,9 @@ export class TaskViewPage {
     });
   }
 
-  markTaskAsCompleted() {
-    this.showAttachments = true;
-  }
+  // markTaskAsCompleted() {
```
Comment (kiranharidas187) [inline src/app/manage-learn/project/task-view/task-view.page.ts:321]: rm

File: src/app/manage-learn/shared/components/add-link-modal/add-link-modal.component.scss [hunk 8]
```diff
@@ -0,0 +1,38 @@
+.custom-popup {
+    width: 100%;
+    height: 100%;
+    display: flex;
+    position: fixed;
+    z-index: 9999;
+    top: 0px;
+    background: #6663;
```
Comment (kiranharidas187) [inline src/app/manage-learn/shared/components/add-link-modal/add-link-modal.component.scss:8]: use var for all color

File: src/app/manage-learn/shared/components/add-link-modal/add-link-modal.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (kiranharidas187) [inline src/app/manage-learn/shared/components/add-link-modal/add-link-modal.component.spec.ts:1]: rm

File: src/app/manage-learn/core/services/project.service.ts [hunk 10]
```diff
@@ -411,4 +411,21 @@ export class ProjectService {
...
+        let link = {
+          name: element,
+          type: 'link',
+          isUploaded: false,
+          url: "",
+        };
+        linkArray.push(link);
+      }
+    });
+    return linkArray.length ? linkArray : links;
```
[the remark this answers is not in the document]
Comment (vishwanath1004) [reply src/app/manage-learn/core/services/project.service.ts:429]: Checked

File: src/app/manage-learn/project/add-file/add-file.page.html [hunk 11]
```diff
@@ -0,0 +1,37 @@
+<ion-content>
+  <app-item-list-header headerLabel="{{'FRMELEMNTS_LBL_ADD_ATTACHMENTS' | translate}}"></app-item-list-header>
```
[the remark this answers is not in the document]
Comment (vishwanath1004) [reply src/app/manage-learn/project/add-file/add-file.page.html:5]: For project and task have same header

File: src/app/manage-learn/project/project-details/project-details.component.ts [hunk 12]
```diff
@@ -263,6 +263,8 @@ export class ProjectDetailsComponent implements OnInit {
     this.db.update(this.projectDetails).then(success => {
       this.projectDetails._rev = success.rev;
       this.taskCount = this.utils.getTaskCount(this.projectDetails);
+      // Project submission is pending so below code commented
```
Comment (kiranharidas187) [inline src/app/manage-learn/project/project-details/project-details.component.ts:266]: rm

File: src/app/manage-learn/project/add-file/add-file.page.ts [hunk 13]
```diff
@@ -0,0 +1,265 @@
...
+  async ionViewWillEnter() {
+    this.headerService.showHeaderWithBackButton();
+  }
+  getProject() {
+    this.db.query({ _id: this.projectId }).then(
+      (success) => {
+        if (success?.docs.length) {
+          this.project = success.docs[0];
+        }
+        this.projectCopy = JSON.parse(JSON.stringify(this.project ));
```
Comment (kiranharidas187) [inline src/app/manage-learn/project/add-file/add-file.page.ts:79]: rm new lines

File: src/app/manage-learn/project/add-file/add-file.page.ts [hunk 14]
```diff
@@ -0,0 +1,265 @@
...
+      (error) => {}
+    );
+  }
+
+  getTask() {
+    this.taskIndex = _.findIndex(this.project.tasks, (item) => {
+      return item._id == this.taskId;
+    });
+    this.task = this.project.tasks[this.taskIndex];
+    this.copyOfTaskDetails = JSON.parse(JSON.stringify(this.task ));
```
Comment (kiranharidas187) [inline src/app/manage-learn/project/add-file/add-file.page.ts:93]: is this still required

File: src/app/manage-learn/shared/components/attachment-card/attachment-card.component.ts [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+  @Input() data: any;
+  @Input() viewOnly: any;
+  @Output() deleteAttachment = new EventEmitter()
+  limit = 2;
+  constructor(
+    private sanitizer: DomSanitizer,
+  ) { }
+
+  ngOnInit() { 
+console.log(this.data,"data");
```
Comment (kiranharidas187) [inline src/app/manage-learn/shared/components/attachment-card/attachment-card.component.ts:19]: rm all logs

**Pull request merged**